master/internal/api_task.go

```diff
@@ -30,3 +30,29 @@ func (a *apiServer) GetTask(
 			errors.Wrapf(err, "error fetching task %s from database", req.TaskId)
 	}
 }
+
+func (a *apiServer) GetGenericTaskConfig(
+	ctx context.Context, req *apiv1.GetGenericTaskConfigRequest,
+) (resp *apiv1.GetGenericTaskConfigResponse, err error) {
+	if err := a.canDoActionsOnTask(ctx, model.TaskID(req.TaskId),
+		expauth.AuthZProvider.Get().CanGetExperimentArtifacts); err != nil {
+		return nil, err
+	}
+
+	t := &taskv1.Task{}
+	switch err := db.Bun().NewSelect().Model(t).
+		Column("config").
+		Where("task_id = ?", req.TaskId).
+		Limit(1).
```

Comment:
this `LIMIT 1` is either redundant (`task_id` is unique in this table, so it can't return more than 1 anyway), or it'll obscure some underlying issue (if you expect a query to return <= 1 and it does not, it'd be better to let it loudly fail in `Scan` call, than to silently hide it).
```suggestion

```